rclcpp_action/src/client.cpp

```diff
@@ -318,14 +318,19 @@ ClientBase::handle_result_response(
   const rmw_request_id_t & response_header,
   std::shared_ptr<void> response)
 {
-  std::lock_guard<std::mutex> guard(pimpl_->result_requests_mutex);
-  const int64_t & sequence_number = response_header.sequence_number;
-  if (pimpl_->pending_result_responses.count(sequence_number) == 0) {
-    RCLCPP_ERROR(pimpl_->logger, "unknown result response, ignoring...");
-    return;
+  std::map<int64_t, ResponseCallback>::node_type pending_result_response;
+  {
+    std::lock_guard<std::mutex> guard(pimpl_->result_requests_mutex);
+    const int64_t & sequence_number = response_header.sequence_number;
+    if (pimpl_->pending_result_responses.count(sequence_number) == 0) {
+      RCLCPP_ERROR(pimpl_->logger, "unknown result response, ignoring...");
+      return;
+    }
+    pending_result_response =
+      pimpl_->pending_result_responses.extract(sequence_number);
```

Comment:
using node_handle and extract method requires `c++17`, so we need to figure out for humble backport in different way.

Reply:
How about using c++11 only so that we can backport Galactic/Humble

```cpp
void
ClientBase::handle_result_response(
  const rmw_request_id_t & response_header,
  std::shared_ptr<void> response)
{
  ResponseCallback response_callback;
  {
    std::lock_guard<std::mutex> guard(pimpl_->result_requests_mutex);
    const int64_t & sequence_number = response_header.sequence_number;
    if (pimpl_->pending_result_responses.count(sequence_number) == 0) {
      RCLCPP_ERROR(pimpl_->logger, "unknown result response, ignoring...");
      return;
    }
    response_callback = std::move(pimpl_->pending_result_responses[sequence_number]);
    pimpl_->pending_result_responses.erase(sequence_number);
  }
  response_callback(response);
}
```

Reply:
yeah i was going to do that for the backport to humble and foxy. but i think we can do this for rolling and iron release since default c++17 applies. I will submit the backport with you suggestion.